### PR TITLE
Add disableCheckbox option to rows

### PR DESCRIPTION
**Pull request by isacoder**
### Description

This is an update request for TypeTable component.
Currently the checkboxes in the rows can't be disabled by developer side.
This update will allow to disable them while still checking the states for the allChecked Checkbox

 ### Considerations on Implementation

I decided to use the logic if at lease one checkbox in rows is disable the checkAll/ unCheckAll will be disable too.
Attaching the reasoning of this logic
Screenshot 2024-10-29 at 14 42 57

I updated the example toggles to use previous as base prevent useEffect forever cycle when updating.

 ### Reviewing/Testing steps
 
Test on storybook `Tables -> Molecules -> LoadingTable`  local using disable 1st checkbox 

Screenshot 2024-10-29 at 14 47 33

### CODE REVIEW

**Review by atomworks — APPROVED** [review]
Code looks good and works well in Storybook, thank you.

I would like to keep the (de)select all option active and just exclude anything that has been disabled. That said it would be useful to keep this as something the developer can actively choose disable for contexts where it makes more sense. Where this feature is needed though at the moment, it makes sense to still allow for select all.

That said, if you want to get this released first so it can be used in projects and follow up with that other change after, that seems like a good plan to me but I shall leave it up to your choice.

**Comment by isacoder** [comment]
> (de)select all option active

Thanks for mentioning.
Currently the logic does not allow it easily. 
`Check All` is internally controlled, but individual checkboxes are controlled from outside. 
If I make that change It will require to give full responsibility of the `check All` logic to the developer and would be breaking changes for previous projects. 

I will merge and if any project requires this functionality we can review the scenario.

**Pull request merged**
